app/src/router/routes/ProtectedRoute.tsx

```diff
@@ -0,0 +1,27 @@
+import ErrorPage from 'components/pages/ErrorPage';
+import Login from 'components/pages/Login';
+import { Role } from 'components/shared/constants/timetable.constants';
+import { IProtectedRouteMeta } from 'components/shared/interfaces/route.interfaces';
+import { IUser } from 'components/shared/interfaces/user.interfaces';
+import React from 'react';
+
+interface IProtectedRouteProps {
+  user: IUser;
+  meta: IProtectedRouteMeta;
+  children: React.ReactNode;
+}
+
+export function ProtectedRoute(props: IProtectedRouteProps) {
+  console.log('Protected Route Props: ', props);
+  if (!props.user) {
+    // Redirect to 401 page if user is not logged in
+    return <Login />;

```

Comment:
We should probably treat this as an actual path change to /login rather than returning the Login page to preserve router history in case the user wants to navigate backwards/forwards.